### ISSUE TITLE
Document kubernetes.io/psp annotation

### ISSUE THREAD

**Issue by disconnect3d** [kubernetes/kubernetes#89940]
Hey,

I have a question regarding validating if PSP is applied, as well as an enhancement request on clarifying this.

It seems to me that when a Pod is validated with a PSP, a `"kubernetes.io/psp"` annotation is added with the name of the used pod security policy as its value. It is not obvious if that mechanism can be relied on, especially since it is not documented in the PSP documentation - https://kubernetes.io/docs/concepts/policy/pod-security-policy/.

However, it can be found to be relied on per the psp/rbac examples readme:
https://github.com/kubernetes/examples/blob/master/staging/podsecuritypolicy/rbac/README.md#testing-access

## "Show me the code"

The `"kubernetes.io/psp"` is defined in [pkg/security/podsecuritypolicy/util/util.go#L29](https://github.com/kubernetes/kubernetes/blob/v1.18.0/pkg/security/podsecuritypolicy/util/util.go#L29) and the annotation is written into the pod object in [plugin/pkg/admission/security/podsecuritypolicy/admission.go#L139](https://github.com/kubernetes/kubernetes/blob/v1.18.0/plugin/pkg/admission/security/podsecuritypolicy/admission.go#L139). I haven't debugged this piece of code nor have I tested if it is invoked in all possible paths, and so on. That's also partially why I want to make this issue: to confirm this detail ;).

## TLDR

**What would you like to be added**:
Documentation about the `kubernetes.io/psp` annotation in Pod Security Policy docs.

**Why is this needed**:
To know how to check if a given Pod was validated by a given PSP. This can be very important when using managed kubernetes services, as some providers add a "privileged" PSP by default, to keep backwards compatibility.

It will also be useful for auditors to have a ground truth for this behaviour and have a place to link to :).

**Comment by disconnect3d** [comment]
Any chance someone confirms it is a valid way to get the applied PSP in all cases? I can send a PR to improve the docs.

Maybe @sttts or @pweil- (via git blame around the linked code)?

**Comment by pweil-** [comment]
Hi @disconnect3d.  The code you point out would run for any pod that passed validation however given the [current state of PSP](https://github.com/kubernetes/enhancements/issues/5#issuecomment-637066475) and the [deprecation policy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/1635-prevent-permabeta) I would defer to @tallclair on whether anything more should be documented.

I also see that the `admission.Attributes` record appears to be given an annotation for the `admit-policy` [here](https://github.com/kubernetes/kubernetes/blob/v1.18.0/plugin/pkg/admission/security/podsecuritypolicy/admission.go#L140) and `validate-policy` [here](https://github.com/kubernetes/kubernetes/blob/v1.18.0/plugin/pkg/admission/security/podsecuritypolicy/admission.go#L168).  Unfortunately I'm not actively involved in the maintenance so I'm not sure if that is a more standardized way to audit what you're looking for vs the deprecated PSP mechanism.  @tallclair can probably help out there too.  Apologies if that just muddies the water.

**Comment by tallclair** [comment]
Yes, this annotation can be relied on as long as PSP is still around.

**Comment by sftim** [comment]
We should document this in https://kubernetes.io/docs/reference/labels-annotations-taints/
/transfer website
